dart/lib/src/sentry_options.dart

```diff
@@ -0,0 +1,210 @@
+  final List<Integration> _integrations = [];
+
+  // TODO: shutdownTimeout, flushTimeoutMillis
+  // https://api.dart.dev/stable/2.10.2/dart-io/HttpClient/close.html doesn't have a timeout param, we'd need to implement manually
+
+  List<Integration> get integrations => List.unmodifiable(_integrations);
+
+  /// Turns debug mode on or off. If debug is enabled SDK will attempt to print out useful debugging
+  /// information if something goes wrong. Default is disabled.
+  bool debug = false;
+
+  /// minimum LogLevel to be used if debug is enabled
+  SentryLevel _diagnosticLevel = defaultDiagnosticLevel;
+
+  set diagnosticLevel(SentryLevel level) {
+    _diagnosticLevel = level ?? defaultDiagnosticLevel;
+  }
+
+  SentryLevel get diagnosticLevel => _diagnosticLevel;
+
+  /// Sentry client name used for the HTTP authHeader and userAgent eg
+  /// sentry.{language}.{platform}/{version} eg sentry.java.android/2.0.0 would be a valid case
+  String sentryClientName;
+
+  /// This function is called with an SDK specific event object and can return a modified event
+  /// object or nothing to skip reporting the event
+  BeforeSendCallback beforeSendCallback;
+
+  /// This function is called with an SDK specific breadcrumb object before the breadcrumb is added
+  /// to the scope. When nothing is returned from the function, the breadcrumb is dropped
+  BeforeBreadcrumbCallback beforeBreadcrumbCallback;
+
+  /// Sets the release. SDK will try to automatically configure a release out of the box
+  String release;
+
+  /// Sets the environment. This string is freeform and not set by default. A release can be
+  /// associated with more than one environment to separate them in the UI Think staging vs prod or
+  /// similar.
+  String environment;
+
+  /// Configures the sample rate as a percentage of events to be sent in the range of 0.0 to 1.0. if
+  /// 1.0 is set it means that 100% of events are sent. If set to 0.1 only 10% of events will be
+  /// sent. Events are picked randomly. Default is null (disabled)
+  double sampleRate;
+
+  /// A list of string prefixes of module names that do not belong to the app, but rather third-party
+  /// packages. Modules considered not to be part of the app will be hidden from stack traces by
+  /// default.
+  final List<String> _inAppExcludes = [];
+
+  List<String> get inAppExcludes => List.unmodifiable(_inAppExcludes);
+
+  /// A list of string prefixes of module names that belong to the app. This option takes precedence
+  /// over inAppExcludes.
+  final List<String> _inAppIncludes = [];
+
+  List<String> get inAppIncludes => List.unmodifiable(_inAppIncludes);
+
+  Transport _transport = NoOpTransport();
+
+  Transport get transport => _transport;
+
+  set transport(Transport transport) =>
+      _transport = transport ?? NoOpTransport();
+
+  // TODO: transportGate, connectionTimeoutMillis, readTimeoutMillis, hostnameVerifier, sslSocketFactory, proxy
```

Comment:
We add on demand? 
```suggestion
```

Reply:
I'll create a card for it

Reply:
https://github.com/getsentry/sentry-dart/projects/1#card-48213567